cirq-core/cirq/protocols/json_serialization_test.py

```diff
@@ -755,6 +755,9 @@ def assert_repr_and_json_test_data_agree(
     inward_only: bool,
     deprecation_deadline: Optional[str],
 ):
+    if deprecation_deadline:
+        # proper_eq can't properly handle equality of deprecated classes
+        return
```

Comment:
Why is this needed? Can you give an example ?

We have successfully deprecated classes in the past without adding this special case, so I'm a bit surprised that it starts failing now.

Reply:
Okay, I think I know what might be going on here. The test would be failing for `GateTabulation`, which is a deprecated class. But it fails because the repr of the class has now changed to `TwoQubitGateTabulation(...)` instead of `GateTabulation(...)` because we are inheriting all methods from the parent class. 

This fix should be:
- Revert the additional if condition added here.
- Delete `cirq-google/cirq_google/json_test_data/GateTabulation.json` and `cirq-google/cirq_google/json_test_data/GateTabulation.repr` s.t. only the corresponding `_inward` files remain. 
- Update `cirq-google/cirq_google/json_test_data/GateTabulation.repr_inward` s.t. `cirq_google.optimizers.two_qubit_gates.gate_compilation.GateTabulation` is replaced with `cirq.TwoQubitGateTabulation` i.e. corresponding to the new repr.  
- Update `cirq-google/cirq_google/json_resolver_cache.py` and replace `'GateTabulation': cirq_google.GateTabulation,` with `'GateTabulation': cirq.TwoQubitGateTabulation,` -- which updates the deserialization logic of the original class.

Reply:
By the way, looking at the __init__.py files, I have noticed that the order of imports for 'names_with_underscores' as opposed to 'NamesWithCamelCases' seems to be a bit arbitrary: e.g. looking at cirq-core/cirq/__init__.py, I can see the ordering applied both ways (camel case before underscores and vice versa). What is the rule here and should a separate ticket be created to normalize the imports everywhere and/or update the lint checker?

Reply:
My reason to suggest reordering was that based on ascii comparison, a capital letter < small letter. so 'A' < 'a' if you sort alphabetically.  +1 to creating a separate issue to track this.

Reply:
Regarding the other suggestion: I tried that, encountered a few shenanigans, was going to disable serialization test for GateTabulation, then found out that whatever issue I had before, it seems to be gone now. So that issue had solved itself.